src/tls/test-cockpit-certificate-ensure.c

```diff
@@ -314,7 +314,7 @@ static const TestCase case_bad_file = {
 static const TestCase case_bad_file2 = {
```

Comment:
The tests need to be extended to cover the cases you mention (empty/truncated file and such). But let's sort out the above questions first.